### PR TITLE
CI for Embedding SDK

### PR DESCRIPTION
**Pull request by deniskaber**
Child of https://github.com/metabase/metabase/issues/40241

### Description

This adds Github actions workflow to build and publish Embedding SDK package.
Job will be triggered if `embedding-sdk-build` label has beed assigned to the PR.

### How to verify

Check previous pipelines - https://github.com/metabase/metabase/actions/workflows/embedding-sdk.yml
Check built package page - https://github.com/metabase/metabase/pkgs/npm/embedding-sdk-react

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: .github/workflows/embedding-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  cancel-in-progress: true
+
+jobs:
+  files-changed:
+    name: Check which files changed
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    outputs:
+      frontend_all: ${{ steps.changes.outputs.frontend_all }}
+      e2e_specs: ${{ steps.changes.outputs.e2e_specs }}
```
Comment (npretto) [inline .github/workflows/embedding-sdk.yml:22]:
```suggestion
```
I don't think we need to save this value

File: .github/workflows/embedding-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        uses: ./.github/actions/prepare-backend
+        with:
+          m2-cache-key: "cljs"
+      - name: Compile CLJS
+        run: yarn run build-release-pure:cljs
+      - name: Build SDK bundle
+        run: yarn run build-release:sdk
+      - name: Generate SDK package.json
+        run: yarn run embedding-sdk:generate-package "$GITHUB_SHA"
+      - name: Generate .npmrc
```
Comment (npretto) [inline .github/workflows/embedding-sdk.yml:51]: Should we explain why we're doing this? Like "Generate .npmrc for custom registry"

File: webpack.embedding-sdk.config.js [hunk 3]
```diff
@@ -39,121 +37,121 @@ const CSS_CONFIG = {
...
-        new BundleAnalyzerPlugin({
-          analyzerMode: "static",
-          reportFilename: BUILD_PATH + "/dist/report.html",
-        }),
-    ].filter(Boolean),
-  };
+  ].filter(Boolean),
 };
+
+// console.log(config);
```
Comment (npretto) [inline webpack.embedding-sdk.config.js:155]:
```suggestion
```

File: webpack.embedding-sdk.config.js [hunk 4]
```diff
@@ -39,121 +37,121 @@ const CSS_CONFIG = {
...
+      banner:
+        "/*\n* This file is subject to the terms and conditions defined in\n * file 'LICENSE.txt', which is part of this source code package.\n */\n",
+    }),
+    new NodePolyfillPlugin(), // for crypto, among others
+    // https://github.com/remarkjs/remark/discussions/903
+    new webpack.ProvidePlugin({
+      process: "process/browser.js",
+    }),
+
+    // TODO: enable this to add types generation (but we also need a separate step to bundle types into a single module)
```
Comment (npretto) [inline webpack.embedding-sdk.config.js:137]: Are we commenting this out because it's not working?

**Review by npretto — COMMENTED** [review]
Left some inline comments

**Review by npretto — COMMENTED** [review]
Left some inline comments

**Review by oisincoveney — APPROVED** [review]
I don't have any complaints, looks good.

File: bin/generate-sdk-package-json.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    ),
+    "utf-8",
+  );
+  const sdkPackageTemplateJsonContent = JSON.parse(sdkPackageTemplateJson);
+
+  const todayDate = new Date().toJSON().slice(0, 10).replaceAll("-", "");
+
+  const mergedContent = {
+    ...sdkPackageTemplateJsonContent,
+    dependencies: filterReactDependencies(mainPackageJsonContent.dependencies),
```
Comment (WiNloSt) [inline bin/generate-sdk-package-json.js:47]: The function `filterReactDependencies` sounds more like it only filter React dependencies. However, it's more like _filtering out_ React dependencies. So I think the name
```suggestion
    dependencies: filterOutReactDependencies(mainPackageJsonContent.dependencies),
```

Or

```suggestion
    dependencies: excludeReactDependencies(mainPackageJsonContent.dependencies),
```

would be more appropriate

**Review by WiNloSt — APPROVED** [review]
I have minor comments, otherwise it looks good.

File: enterprise/frontend/src/embedding-sdk/components/public/QueryVisualization.tsx [hunk 6]
```diff
@@ -119,7 +121,7 @@ export const QueryVisualizationSdk = ({
 
   return (
     <LoadingAndErrorWrapper
-      className="flex-full full-width"
+      className={cx(CS.flexFull, CS.fullWidth)}
```
[the remark this answers is not in the document]
Comment (WiNloSt) [reply enterprise/frontend/src/embedding-sdk/components/public/QueryVisualization.tsx:124]: Ok that's fair

**Pull request merged**